### ISSUE TITLE
Increase default server memory size

### ISSUE THREAD

**Issue by madskjeldgaard** [supercollider/supercollider#5558]
## Motivation

The default memory size for the server is currently 8192, about 8mb. This is extremely conservative by today's standards and I think this leads to some confusion for new(er) users once they want to start loading sound files into buffers for example, which will quickly lead to those 8 megabytes being exhausted. The user then has to figure out the issue, find the startup file / or add the setting to the project at hand and then be able to do what the user wanted to. I think the default should be more modern and then for users who want an extremely small memory footprint they can set the memory size in a startup file, instead of vice-versa as it is now.

related: https://github.com/supercollider/supercollider/issues/5557

## Description of Proposed Feature
I think the value of the memory size could reasonably be raiseed to 128mb or 256mb without causing too much trouble. Not sure what exact sweetspot here is but I guess a lot of laptops nowadays have approximately 4-8GB of ram? 

## Plan for Implementation

Change this line:
https://github.com/supercollider/supercollider/blob/e4743da23948f6ce1cfe7750273814d0f489cc20/SCClassLibrary/Common/Control/Server.sc#L74

**Comment by madskjeldgaard** [comment]
Somewhat related: If there is consensus for changing this we could perhaps also take a look at defaults for server options `numWireBufs`, `maxSynthDefs`, `maxNodes` as well.

**Comment by alln4tural** [comment]
I'm not against raising the default, but note that loaded buffers do not count towards the total: http://doc.sccode.org/Classes/ServerOptions.html#-memSizeCheers,eddihttps://alln4tural.bandcamp.comOn 31/08/2021, 20:26 mads kjeldgaard ***@***.***> wrote:

   
  Motivation 
  The default memory size for the server is currently 8192, about 8mb. This is extremely conservative by today's standards and I think this leads to some confusion for new(er) users once they want to start loading sound files into buffers for example, which will quickly lead to those 8 megabytes being exhausted. The user then has to figure out the issue, find the startup file / or add the setting to the project at hand and then be able to do what the user wanted to. I think the default should be more modern and then for users who want an extremely small memory footprint they can set the memory size in a startup file, instead of vice-versa as it is now. 
  related: #5557 
  Description of Proposed Feature 
  I think the value of the memory size could reasonably be raiseed to 128mb or 256mb without causing too much trouble. Not sure what exact sweetspot here is but I guess a lot of laptops nowadays have approximately 4-8GB of ram? 
  Plan for Implementation 
  Change this line: https://github.com/supercollider/supercollider/blob/e4743da23948f6ce1cfe7750273814d0f489cc20/SCClassLibrary/Common/Control/Server.sc#L74

**Comment by dyfer** [comment]
Yes, as @alln4tural said `memSize` does _not_ count for buffer allocated memory. It is used e.g. when a delay line needs to allocate internal memory when the synth starts.
8 MB is somewhat conservative by today's levels, but we don't need that much typically (depending on the use case). If we raise this I think it doesn't need to go higher than 64 (or maybe 128) MB.

**Comment by telephon** [comment]
… but there may be a limitation: https://github.com/supercollider/supercollider/issues/5469#issuecomment-909333911

**Comment by madskjeldgaard** [comment]
> I'm not against raising the default, but note that loaded buffers do not count towards the total: http://doc.sccode.org/Classes/ServerOptions.html#-memSize Cheers, eddi https://alln4tural.bandcamp.com

Thanks for the correction. It was  a bad example !

**Comment by madskjeldgaard** [comment]
> In SuperDirt, we set it to `8192 * 32`, otherwise we get to the limit relatively quickly.

I think this is a reasonable default but perhaps half of this is a little bit more conservative? I don't know. Just opened up Reaper for example and it uses ~126 mb for the whole program and ardour is around 190mb it seems. Both on Linux. That said, I think I might lean more towards this default of 32 * 8192 which is probably future proof.

**Comment by redFrik** [comment]
contrary opinion... if the user is doing something unwise or uninformed like allocating delay lines inside a grain synth, i think it's rather good that the system pushes back.  and in the few cases you really need that much memory it's easy to just allocate more manually (like we do now).  it can be seen as an opportunity to learn about memory, signal routing and how to be resourceful.

**Comment by muellmusik** [comment]
2 cents: I'd prefer less confusion and fails at the expense of opportunities for character building. 32*8192 sounds good.

**Comment by jamshark70** [comment]
> Just opened up Reaper for example and it uses ~126 mb for the whole program and ardour is around 190mb it seems.

Better to report both VIRT and RSS:

scsynth: I have `s.options.memSize = 32*8192` which is even more than proposed here -- VIRT = 944.6M, RSS = 131.9M (per XFCE Task Manager).

Reaper: VIRT = 679.5M, RSS = 192.4M

Virtual memory 944.6M vs 679.5M is not really outrageous IMO. I think `32* 8192` is OK.

**Comment by madskjeldgaard** [comment]
Somewhat related: should the `maxSynthDefs` and `numWireBufs` options also have their defaults increased? They seem like slightly arbitrary limitations at the moment

**Comment by dyfer** [comment]
In the interest of moving this forward, I'd like to ping servers' maintainers: @elgiano @Spacechild1 (even thought this is set in sclang). How should we proceed?

**Comment by Spacechild1** [comment]
> even thought this is set in sclang

Note that the default `ServerOptions` values often just mirror those in `WorldOptions` (in `SC_WorldOptions.h`).

I don't really have a strong opinion on this. Generally, I agree that 8 MB is too low for many use cases. 256 MB is quite a lot of memory, but I guess in the case of sclang, where you typically only have a single scsynth instance, this would be acceptable. In that case, I would probably keep 8192 kB on the server side (`WorldOptions`) and only increase it on the sclang side (`ServerOptions`).

**Comment by capital-G** [comment]
Developer meeting @ 2025-08-25 suggested to bump the values in sclang and scsynth and supernova

value | old | now
--- | --- | ---
`memSize` | 8192 KB |  256 MB
`maxSynthDefs` | 1024 | 8192
`numWireBufs` | 64 | 512

Please do a PR @JordanHendersonMusic  :)

Ping TidalCycles and https://github.com/supercollider/supercollider/blob/fed154ee8277776f92fdcca031d2c45ffb891304/HelpSource/Help.schelp#L34

**Comment by telephon** [comment]
https://codeberg.org/musikinformatik/SuperDirt/pulls/321
